integration/test/test_cpu_characterization.py

```diff
@@ -0,0 +1,207 @@
+
+        #TODO: dropping max freq for max many core freq?
+        #cls._cpu_max_freq = geopm_test_launcher.geopmread("CPU_FREQUENCY_MAX_AVAIL board 0")
+        cls._cpu_max_freq = geopm_test_launcher.geopmread("MSR::TURBO_RATIO_LIMIT:MAX_RATIO_LIMIT_7 board 0")
+        cls._uncore_max_freq = geopm_test_launcher.geopmread("CPU_UNCORE_FREQUENCY_MAX_CONTROL board 0")
+
+        cls._cpu_min_freq = geopm_test_launcher.geopmread("CPU_FREQUENCY_MIN_AVAIL board 0")
+        cls._uncore_min_freq = geopm_test_launcher.geopmread("CPU_UNCORE_FREQUENCY_MIN_CONTROL board 0")
+
+        node_count = 1
+
+        mach = machine.init_output_dir('.')
+
+        def launch_helper(experiment_type, experiment_args, app_conf, experiment_cli_args):
+            output_dir = experiment_args.output_dir
+            if output_dir.exists() and output_dir.is_dir():
+                shutil.rmtree(output_dir)
+
+            experiment_type.launch(app_conf=app_conf, args=experiment_args,
+                                   experiment_cli_args=experiment_cli_args)
+
+        #####################
+        # Setup Common Args #
+        #####################
+        geopm_test_launcher.geopmwrite("CPU_FREQUENCY_MAX_CONTROL board 0 {}".format(cls._cpu_max_freq))
+        geopm_test_launcher.geopmwrite("CPU_UNCORE_FREQUENCY_MIN_CONTROL board 0 {}".format(cls._uncore_min_freq))
+        geopm_test_launcher.geopmwrite("CPU_UNCORE_FREQUENCY_MAX_CONTROL board 0 {}".format(cls._uncore_max_freq))
+
+        experiment_args = SimpleNamespace(
+            node_count=node_count,
+            trial_count=3,
+            cool_off_time=3,
+            enable_traces=False,
+            enable_profile_traces=False,
+            verbose=False,
+        )
+
+        # Arithmetic Intensity Benchmark
+        # Only benchmarks 1 & 16 are used to reduce characterization time
+        aib_app_conf = arithmetic_intensity.ArithmeticIntensityAppConf(
+            ['--slowdown=1',
+             '--base-internal-iterations=10',
+             '--iterations=5',
+             f'--floats={1<<21}', #quick characterization, use 26 for slower & more accurate
+             '--benchmarks=1 16'],
+            mach,
+            run_type='sse',
+            ranks_per_node=None,
+            distribute_slow_ranks=False)
+
+        # use a two step characterization process to reduce runtime
+        # Uncore sweep with core = base freq first
+        # then core sweep with uncore at the uncore fe
+
+        ##################################
+        # Uncore frequency sweep at base #
+        ##################################
+        cls._aib_uncore_freq_sweep_dir = Path(os.path.join('test_cpu_characterization_output', 'uncore_frequency_sweep'))
+        experiment_args.output_dir = cls._aib_uncore_freq_sweep_dir
+        experiment_args.min_frequency = cls._cpu_base_freq
+        experiment_args.max_frequency = cls._cpu_base_freq
+        experiment_args.step_frequency = 1e8 #TODO: use step freq signal
```

Comment:
`mach` is already in scope here, so I think this can use `mach.frequency_step()`